core/app/models/spree/promotion/rules/user_role.rb

```diff
@@ -0,0 +1,26 @@
+module Spree
+  class Promotion
+    module Rules
+      class UserRole < PromotionRule
+        has_many :promotion_rule_roles,
+          class_name: 'Spree::PromotionRuleRole',
+          foreign_key: :promotion_rule_id,
+          dependent: :destroy
+
+        has_many :roles,
+          through: :promotion_rule_roles,
+          class_name: 'Spree::Role'
+
+        accepts_nested_attributes_for :roles
+
+        def applicable?(promotable)
+          promotable.is_a?(Spree::Order)
+        end
+
+        def eligible?(order, options = {})
+          (roles & order.user.spree_roles).any? if order.user
```

Comment:
I guess perhaps I have been worn down by a Marketing departments that can and will hang themselves whenever I give them even the shortest bit of rope (As well, I've scaled my share of apps and am very sensitive to bloat in Promotions.)

Having said that, it seems to me that conceptually speaking, if Roles are a first-class entity in the domain design, then having a Promotion rule associated with Roles is fine too.